pkg/controller/customdomain/customdomain_controller.go

```diff
@@ -170,80 +154,97 @@ func createOrAddCustomAppsDomain(r *ReconcileCustomDomain, reqLogger logr.Logger
-		reqLogger.Error(err, "Error getting dns.config/cluster")
-		return err
+		reqLogger.Error(err, fmt.Sprintf("Error getting dns.config/%s", dnsConfigName))
+		return reconcile.Result{}, err
 	}
 
 	// set the ingress domain to be a subdomain under the cluster's installed basedomain
 	// such that the record is added to the zone and external DNS can point to it
 	ingressDomain := fmt.Sprintf("%s.%s", instance.Name, dnsConfig.Spec.BaseDomain)
 	ingressName := instance.Name
 
-	// create or update ingresscontrollers.openshift.io/custom
+	// create new ingresscontrollers.openshift.io
 	customIngress := &operatorv1.IngressController{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{
 		Namespace: ingressOperatorNamespace,
 		Name:      ingressName,
 	}, customIngress)
-	customIngress.Name = ingressName
-	customIngress.Namespace = ingressOperatorNamespace
-	customIngress.Spec.Domain = ingressDomain
-	if customIngress.Spec.DefaultCertificate != nil {
-		customIngress.Spec.DefaultCertificate.Name = secretName
-	} else {
-		customIngress.Spec.DefaultCertificate = &corev1.LocalObjectReference{Name: secretName}
-	}
 	if err != nil {
-		err = r.client.Create(context.TODO(), customIngress)
-		if err != nil {
-			reqLogger.Error(err, "Error creating custom ingress controller")
-			return err
+		if errors.IsNotFound(err) {
+			customIngress.Name = ingressName
+			customIngress.Namespace = ingressOperatorNamespace
+			customIngress.Spec.Domain = ingressDomain
+			if customIngress.Spec.DefaultCertificate != nil {
+				customIngress.Spec.DefaultCertificate.Name = secretName
+			} else {
+				customIngress.Spec.DefaultCertificate = &corev1.LocalObjectReference{Name: secretName}
+			}
+			err = r.client.Create(context.TODO(), customIngress)
+			if err != nil {
+				reqLogger.Error(err, fmt.Sprintf("Error creating ingresscontroller %s in %s namespace", ingressName, ingressOperatorNamespace))
+				return reconcile.Result{}, err
+			}
+		} else {
+			reqLogger.Error(err, fmt.Sprintf("Error getting ingresscontroller %s in %s namespace", ingressName, ingressOperatorNamespace))
+			return reconcile.Result{}, err
 		}
-	} else {
-		err = r.client.Update(context.TODO(), customIngress)
-		if err != nil {
-			reqLogger.Error(err, "Error updating custom ingress controller")
-			return err
+	}
+	reqLogger.Info(fmt.Sprintf("The ingresscontroller %s already exists in the %s namespace", ingressName, ingressOperatorNamespace))
+
+	// Obtain the dnsRecord to set in the CR status for final completion, requeue if not available
+	dnsRecord := &operatoringressv1.DNSRecord{}
+	dnsRecordName := fmt.Sprintf("%s-wildcard", instance.Name)
+	err = r.client.Get(context.TODO(), types.NamespacedName{
+		Namespace: ingressOperatorNamespace,
+		Name:      dnsRecordName,
+	}, dnsRecord)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// requeue and wait for record
+			return reconcile.Result{Requeue: true, RequeueAfter: time.Duration(requeueWaitMinutes) * time.Minute}, nil
```

Comment:
Suggest making `requeueWaitTime` a `time.Duration` instead of multiplying it out here. This [bit us in aws-account-operator](https://github.com/openshift/aws-account-operator/pull/496) -- totally human error, but would have been a harder human error to make if the WaitTime had already been a Duration.